modules/data.atmosphere/R/download.ICOS.R

```diff
@@ -0,0 +1,234 @@
+#' Download ICOS Ecosystem data products
+#'
+#' Currently available products: 
+#' Drought-2018 ecosystem eddy covariance flux product https://www.icos-cp.eu/data-products/YVR0-4898
+#' ICOS Final Fully Quality Controlled Observational Data (Level 2) https://www.icos-cp.eu/data-products/ecosystem-release
+#'
+#'
+#' @param sitename ICOS id of the site. Example - "BE-Bra"
+#' @param outfolder path to the directory where the output file is stored. If specified directory does not exists, it is created.
+#' @param start_date start date of the data request in the form YYYY-MM-DD
+#' @param end_date end date area of the data request in the form YYYY-MM-DD
+#' @param product ICOS product to be downloaded. Currently supported options: "Drought2018", "ETC"
+#' @param overwrite should existing files be overwritten. Default False.
+#' @return information about the output file
+#' @export
+#' @examples
+#' \dontrun{
+#' download.ICOS("FI-Sii", "/home/carya/pecan", "2016-01-01", "2018-01-01", product="Drought2018")
+#' }
+#' @author Ayush Prasad
+#'
+download.ICOS <-
+  function(sitename,
+           outfolder,
+           start_date,
+           end_date,
+           product,
+           overwrite = FALSE) {
+    download_file_flag <- TRUE
+    extract_file_flag <- TRUE
+    
+    
+    if (tolower(product) == "drought2018") {
+      # construct output CSV file name
+      output_file_name <-
```

Comment:
I think this is a logical construction, but should we rather leave start year in this filename as a wildcard or something like that? Take the vignette example, for FI-Hyy the unzipped file ends up being `1996-2018_beta-3.csv`, but your filename stays as `2016-2018_beta-3.csv`

On L79 try different checks to see if csv file exists, then downstream if `zipped_csv_name` is different than `output_file_name`, overwrite it with `zipped_csv_name` maybe. This way DB record will be accurate